app/vite.config.js

```diff
@@ -6,10 +6,35 @@ import { defineConfig } from "vite";
 import { VitePWA } from "vite-plugin-pwa";
 import { viteStaticCopy } from "vite-plugin-static-copy";
 
-// import path from "path" causes eslint to crash for some reason
+// import from "path" and "fs" causes eslint to crash for some reason
 const path = require("path");
+const { readFileSync, writeFileSync } = require("fs");
 
-export default defineConfig({
+function prodScriptPlugin(command) {
```

Comment:
this is a great idea but the script modifies `templates/base.html` in-place, so the prod version might get checked into git instead of just being a prod build output.

Maybe save the original dev file to `*.htmdev` (some file extension thats not `.html` so that it doesnt get bundled) in this hook and then have a `closeBundle()` hook that restores all `*.htmdev` files to `.html` (and overwrite the prod file) when everything is completed?

Reply:
actually wait doing [the htmdev thing](https://github.com/igneous-labs/htmx-axum-template/pull/1#discussion_r1308207130) might be bad because if you cancel the build halfway the `closeBundle()` hook doesnt run. Maybe we flip that and change the `inputOptions` for rollup to use `*.htmprod` files instead of `*.html` files (not sure if rollup requires a strict `.html` file extension) and then we gitignore `.htmprod` files

Reply:
ill see what we can do

Reply:
heres the process i managed to make:
1. copy over all .html files to `tempHtmlFiles`
2. process them (replace dev scripts with prod ones)
3. build them to `dist/tempHtmlFiles`
4. move from `dist/tempHtmlFiles` to just `dist` and remove `dist/tempHtmlFiles` folder in `postbuild` script
5. `remove `tempHtmlFiles` (from step 1) in vite `builteEnd` script

Reply:
unfortunately seems like we cant do step 4 inside of `vite build` cause if `tempHtmlFiles` and `dist/tempHtmlFiles` doesnt exist before the build script - it won't work (my guess is either vite or rollup is "indexing" files before the build and cant work with files that were created during it)